loky/backend/_thread_pool_limiters.py

```diff
@@ -0,0 +1,506 @@
+
+    def _check_prefix(self, library_basename, filename_prefixes):
+        for prefix in filename_prefixes:
+            if library_basename.startswith(prefix):
+                return prefix
+        return None
+
+    def _include_modules(self, module_info, prefix, prefixes, user_api):
+        return prefix is not None and (prefix in prefixes or
+                                       module_info['user_api'] in user_api)
+
+    def _make_module_info(self, module_path, module_info, prefix):
+        """Make a dict with the information from the module."""
+        module_path = os.path.normpath(module_path)
+        clib = ctypes.CDLL(module_path)
+        internal_api = module_info['internal_api']
+        set_func = getattr(clib, MAP_API_TO_FUNC[internal_api]['set'],
+                           lambda n_thread: None)
+        get_func = getattr(clib, MAP_API_TO_FUNC[internal_api]['get'],
+                           lambda: None)
+        module_info = module_info.copy()
+        module_info.update(clib=clib, set=set_func, get=get_func,
+                           module_path=module_path, prefix=prefix,
+                           version=self.get_version(clib, internal_api))
+        return module_info
+
+    def _get_module_info_from_path(self, module_path, prefixes, user_api):
+        module_name = os.path.basename(module_path).lower()
+        for info in SUPPORTED_IMPLEMENTATIONS:
+            prefix = self._check_prefix(module_name, info['filename_prefixes'])
+            if self._include_modules(info, prefix, prefixes, user_api):
+                self.cls_thread_locals._modules.append(
+                    self._make_module_info(module_path, info, prefix))
+
+    def _find_modules_with_clibs_dl_iterate_phdr(self, prefixes, user_api):
+        """Loop through loaded libraries and return binders on supported ones
+
+        This function is expected to work on POSIX system only.
+        This code is adapted from code by Intel developper @anton-malakhov
+        available at https://github.com/IntelPython/smp
+
+        Copyright (c) 2017, Intel Corporation published under the BSD 3-Clause
+        license
+        """
+
+        libc = self._get_libc()
+        if not hasattr(libc, "dl_iterate_phdr"):  # pragma: no cover
+            return []
+
+        self.cls_thread_locals._modules = []
+
+        # Callback function for `dl_iterate_phdr` which is called for every
+        # module loaded in the current process until it returns 1.
+        def match_module_callback(info, size, data):
+
+            # Get the path of the current module
+            module_path = info.contents.dlpi_name
+            if module_path:
+                module_path = module_path.decode("utf-8")
+
+                # Store the module in cls_thread_locals._module if it is
+                # supported and selected
+                self._get_module_info_from_path(module_path, prefixes,
+                                                user_api)
```

Comment:
I think `_get_module_info_from_path` should be renamed `_collect_module_info_from_path` and take a `collected_modules` local variable as argument and we could get rid of the `self.cls_thread_locals._modules` class local attribute which feels like a hack.